discovery-provider/src/tasks/aggregates/index_aggregate_track.py

```diff
@@ -0,0 +1,155 @@
+import logging
+
+from src.tasks.aggregates import try_updating_aggregate_table, update_aggregate_table
+from src.tasks.celery_app import celery
+from src.tasks.index_aggregate_user import get_latest_blocknumber
+
+logger = logging.getLogger(__name__)
+
+AGGREGATE_TRACK = "aggregate_track"
+
+# UPDATE_AGGREGATE_TRACK_QUERY
+# Get a lower bound blocknumber to check for new save/repost activity
+# Find a subset of tracks that have changed after that blocknumber
+# For that subset of tracks calculate the play/repost total counts
+# Insert that count for new users or update an existing row
+UPDATE_AGGREGATE_TRACK_QUERY = """
+    WITH new_track AS (
+        SELECT
+            t.track_id AS track_id
+        FROM
+            tracks t
+        WHERE
+            t.is_current IS TRUE
+            AND t.blocknumber > :prev_blocknumber
+        GROUP BY
+            t.track_id
+        UNION
+        ALL (
+            SELECT
+                r.repost_item_id AS track_id
+            FROM
+                reposts r
+            WHERE
+                r.is_current IS TRUE
+                AND r.repost_type = 'track'
+                AND r.is_delete IS FALSE
+                AND r.blocknumber > :prev_blocknumber
+            GROUP BY
+                r.repost_item_id
+        )
+        UNION
+        ALL (
+            SELECT
+                s.save_item_id AS track_id
+            FROM
+                saves s
+            WHERE
+                s.is_current IS TRUE
+                AND s.save_type = 'track'
+                AND s.is_delete IS FALSE
+                AND s.blocknumber > :prev_blocknumber
+            GROUP BY
+                s.save_item_id
+        )
+    )
+    AND deleted_tracks AS (
+        DELETE FROM
+            aggregate_track a
+        WHERE
+            a.track_id IN (
+                SELECT
+                    track_id
+                FROM
+                    tracks d
+                WHERE
+                    t.is_delete IS TRUE
+                    AND d.blocknumber > :prev_blocknumber
+            )
+    )
```

Comment:
@dmanjunath @isaacsolo @jowlee How does this additional CTE look?

I assume we don't have to have to call it anywhere since it will auto delete for us when it creates the CTE itself?

Reply:
this code looks good to me. we should make sure to test this case to ensure it actually works

Reply:
Looks like there's a syntax error with the AND but yeah I think this should work. Adding a case in the integration tests would be good.

Reply:
Made a few changes to the query to get the new test case to work: 0f34ab46d54317a657d24c39132a0420732462f1.

The changes are that:

* we filter out `t.is_delete IS TRUE` on our `UNION` and `WHERE` statements, since we won't be looking at these going forward.
* we filter our new CTE for `d.is_current IS TRUE` to ensure undeleted tracks are not deleted from aggregate_track.

@dmanjunath @isaacsolo Can you confirm these changes are sane?